aten/src/ATen/native/Activation.cpp

```diff
@@ -725,36 +725,37 @@ Tensor leaky_relu_backward(
   return iter.output();
 }
 
-std::tuple<Tensor, Tensor> log_sigmoid_forward_cpu(const Tensor& input) {
-  // FIXME: do these actually need to be zeros_like or can they be empty_like?
-  auto result = at::zeros_like(input, at::MemoryFormat::Contiguous);
-  auto buffer = at::zeros_like(input, at::MemoryFormat::Contiguous);
-  log_sigmoid_cpu_stub(kCPU, result, buffer, input.contiguous());
+std::tuple<Tensor, Tensor> log_sigmoid_forward(const Tensor& input) {
+  auto result = at::empty_like(input, at::MemoryFormat::Contiguous);
+  auto buffer = at::empty_like(input, at::MemoryFormat::Contiguous);
```

Comment:
Correct me if I'm wrong, but you allocate buffer regardless of device type, however you are not going to use this buffer on CUDA, am I right?